src/CppAst/CppModelBuilder.cs

```diff
@@ -1975,41 +2065,87 @@ protected void WarningUnhandled(CXCursor cursor, CXCursor parent)
+                        }
+                        break;
+
+                    case CXTemplateArgumentKind.CXTemplateArgumentKind_Integral:
+                        {
+                            var value = templateArg.AsIntegral;
+                            var templateParam = TryToCreateTemplateParameters(cursor.GetTemplateParameter(0, (uint)templateIndex), data) as CppTemplateParameterNonType;
+                            if (templateParam != null)
+                            {
+                                var templateCppType = new CppTemplateParameterType(templateParam.Name);
+                                specializedTemplateCppTypes.Add(new CppTemplateArgument(templateCppType, value));
+                            }
+                        }
+                        break;
+
+                    case CXTemplateArgumentKind.CXTemplateArgumentKind_Expression:
+                        {
+                            var expression = VisitExpression(templateArg.AsExpr, data);
+                            if (expression.Kind == CppExpressionKind.IntegerLiteral)
+                            {
+                                var templateParam = TryToCreateTemplateParameters(cursor.GetTemplateParameter(0, (uint)templateIndex), data) as CppTemplateParameterNonType;
+                                if (templateParam != null)
+                                {
+                                    var templateCppType = new CppTemplateParameterType(templateParam.Name);
+                                    var literal = expression as CppLiteralExpression;
+                                    specializedTemplateCppTypes.Add(new CppTemplateArgument(templateCppType, Int32.Parse(literal.Value)));
+                                }
+                            }
+                            else if (expression.Kind == CppExpressionKind.Unexposed)
+                            {
+                                var templateParam = TryToCreateTemplateParameters(cursor.GetTemplateParameter(0, (uint)templateIndex), data) as CppTemplateParameterNonType;
+                                if (templateParam != null)
+                                {
+                                    var templateCppType = new CppTemplateParameterType(templateParam.Name);
+                                    var raw = expression as CppRawExpression;
+                                    Int32 argument;
+                                    var isInteger = Int32.TryParse(raw.Text, out argument);
+                                    if (isInteger)
+                                    {
+                                        specializedTemplateCppTypes.Add(new CppTemplateArgument(templateCppType, argument));
+                                    }
+                                    else
+                                    {
+                                        specializedTemplateCppTypes.Add(new CppTemplateArgument(templateCppType, raw.Text));

```

Comment:
This seems concerning. What is usually in `raw.Text`?

Reply:
I fixed that, e.g. when the template parameter is really a text, it is not really a specialized template argument, but a template parameter (which is parsed anyhow)